docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,409 @@
+
+# Improved schema for operator backed services
+
+This proposal is about improving the experience for a user of the operator backed services. Currently a user doesn't know exaustively which fields are available in a CR as we depend on optional metadata present in CRDs, to improve this we are considering using the metadata available from the cluster about the CRDs.
+Getting that metadata from the cluster is challenging because a normal cluster user ( plain vanilla ) doesn't have access to that metadata so we are gonna follow a 3 tiered approach
+
+- check if the user has access to the CRD, if so then use that.
+- get the `swagger.json` from the openapi endpoint provided by kubernetes. a sample can be found here https://gist.github.com/girishramnani/f29949d5cb8c6547102776437e05ac19
+- finally if we cannot get the openapi data as well then use the `ClusterServiceVersion` to get the parameters.
+
+This change would affect multiple service commands and the changes are described briefly below -
+- `odo catalog describe service` should include the metadata fields with description so the users can provide then when doing `odo service create`
```

Comment:
To just make sure that we don't forget this:
Every command should have working and updated `-o json`.

Reply:
sure, but we would have different json output structurally for when user has access to `CRD` vs just `CSV`

Reply:
> sure, but we would have different json output structurally for when user has access to `CRD` vs just `CSV`

that is not good. We need to align  it

Reply:
fixed